src/tokens2.jl

```diff
@@ -66,7 +66,16 @@ end
          ii[2].address == 1?                       2 :
          ii[2].address == 2?                       3 : 1
 
+"""
+    compare(m::SAContainer, s::IntSemiToken, t::IntSemiToken)
 
+### Args:
+* `m`: SortedDict or SortedMultiDict.
+* `s`: Semitoken
+* `t`: Semitoken
+
+This function determines the relative positions of the data items indexed by (m,s) and (m,t) in the sorted order.
```

Comment:
Daniel and Milan,

I'm not completely clear on what you are doing in this thread, but the 'compare' function is documented in the DataStructures.jl documentation:

compare(sc,st1,st2)
    Here, st1 and st2 are semitokens for the same container sc; this function determines the relative positions of the data items indexed by (sc,st1) and (sc,st2) in the sorted order. The return value is -1 if (sc,st1) precedes (sc,st2), 0 if they are equal, and 1 if (sc,st1) succeeds (sc,st2). This function compares the tokens by determining their relative position within the tree without dereferencing them. For SortedDict it is mostly equivalent to comparing deref_key((sc,st1)) to deref_key((sc,st2)) using the ordering of the SortedDict except in the case that either (sc,st1) or (sc,st2) is the before-start or past-end token, in which case the deref operation will fail. Which one is more efficient depends on the time-complexity of comparing two keys. Similarly, for SortedSet it is mostly equivalent to comparing deref((sc,st1)) to deref((sc,st2)). For SortedMultiDict, this function is not equivalent to a key comparison since two items in a SortedMultiDict with the same key are not necessarily the same item. Time: O(log n)